crates/lox-io/src/ndm/common.rs

```diff
@@ -0,0 +1,1624 @@
+    pub ballistic_coeff: Option<BallisticCoeffType>,
+    #[serde(rename = "THRUST_ACCELERATION")]
+    pub thrust_acceleration: Option<Ms2Type>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct AltType {
+    #[serde(rename = "$text")]
+    pub base: AltRange,
+    #[serde(rename = "@units")]
+    pub units: Option<LengthUnits>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct BallisticCoeffType {
+    #[serde(rename = "$text")]
+    pub base: NonNegativeDouble,
+    #[serde(rename = "@units")]
+    pub units: Option<BallisticCoeffUnitsType>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct LatType {
+    #[serde(rename = "$text")]
+    pub base: LatRange,
+    #[serde(rename = "@units")]
+    pub units: LatLonUnits,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct LonType {
+    #[serde(rename = "$text")]
+    pub base: LonRange,
+    #[serde(rename = "@units")]
+    pub units: LatLonUnits,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedType {
+    #[serde(rename = "COMMENT")]
+    pub comment_list: Vec<String>,
+    #[serde(rename = "USER_DEFINED")]
+    pub user_defined_list: Vec<UserDefinedParameterType>,
+}
+
+#[derive(
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    serde::Deserialize,
+    serde::Serialize,
+    lox_derive::KvnDeserialize,
+)]
+#[serde(default)]
+pub struct UserDefinedParameterType {
+    #[serde(rename = "$text")]
+    pub base: String,
+    #[serde(rename = "@parameter")]
+    pub parameter: String,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionType {}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionRateType {}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct QuaternionDotType {
+    #[serde(rename = "$text")]
+    pub base: f64,
+    #[serde(rename = "@units")]
+    pub units: Option<QuaternionDotUnits>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationAngleType {
+    #[serde(rename = "rotation1")]
+    pub rotation1: RotationAngleComponentType,
+    #[serde(rename = "rotation2")]
+    pub rotation2: RotationAngleComponentType,
+    #[serde(rename = "rotation3")]
+    pub rotation3: RotationAngleComponentType,
+}
+
+#[derive(Clone, Debug, Default, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct RotationAngleComponentTypeold {
```

Comment:
```suggestion
pub struct RotationAngleComponentTypeOld {
```

Reply:
Is this still used, or is it a refactoring leftover?

Reply:
This is generated. It exists in the xsd file but I don't see it being used in any of the messages, not even in the ones that I haven't implemented.